File: FractalMandelBrot/MandelBrot.cpp
#include "pch.h"
#include "MandelBrot.h"
#include <complex>

using namespace std;

namespace FractalMandelbrot
{

   MandelBrot::MandelBrot(int inMaxIteration) : mMaxIteration(inMaxIteration)
   {
   }


   MandelBrot::~MandelBrot()
   {
   }

   int MandelBrot::getIteration(double x, double y, int inMaxiterations)
   {
      complex<double> z = 0;
      complex<double> c(x,y);

      int iterations = 0;

      while(iterations < inMaxiterations)
      {
         z = z * z + c;
         if (abs(z) > 2)
         {
            break;
         }
         ++iterations;
      }

      return iterations;
   }

}

File: SharpMandelbrot/MandelBrot.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Numerics;

namespace SharpMandelbrot
{
    class MandelBrot
    {
        //private:
        int mMaxIteration = 100;
        //public:
        public MandelBrot(int inMaxIteration)
        {
            mMaxIteration = inMaxIteration;
        }



        public static int getIteration(double x, double y, int inMaxiterations)
        {
            Complex z = new Complex(0,0);
            Complex c = new Complex(x, y);

            int iterations = 0;

            while (iterations < inMaxiterations)
            {
                z = z * z + c;
                double lMag = Complex.Abs(z);
                if (lMag > 2)
                {
                    break;
                }
                ++iterations;
            }

            return iterations;
        }
    }
}
